### ISSUE TITLE
Add-DbrServerToInventory: can't determine port number

### ISSUE THREAD

**Issue by amtwo** [sqlcollaborative/dbareports#1]
Depending on setup, when adding a new instance, `Connect-SqlServer` might not connect over TCP. When this, happens `Add-DbrServerToInventory` can't determine the port number (it's null):
`SELECT local_tcp_port FROM sys.dm_exec_connections WHERE session_id = @@SPID`

In particular, this happens if you run Add-DbrServerToInventory from the console of the server running that SQL instance. The default behavior in that scenario is to connect using Shared Memory instead of TCP, even if TCP is available.

When checking for TCP port, we should explicitly prepend TCP: to the server name when we connect. By connecting to TCP:ServerName we can ensure that querying `sys.dm_exec_connections` for local_tcp_port will return a valid value.

**Comment by SQLDBAWithABeard** [comment]
Question - Is the instancelist table the correct place for Port? This was originally included for a specific environment with none-default ports and hidden instances requiring explicit ports to be specified for conneciton string. If this is too unusual a practice then we could remove the Port from the instance list and add it to the SQLInfo table where it probably should belong.

However this will have an impact on any connections that need to use none standard ports and don't have sql browser running

**Comment by ClaudioESSilva** [comment]
I think yes!

When we have multiple instances on the same server,  the port is different
even if the ip is the same.

I like to see the port there.  That way we can connect using named instance
or "ip,  Port".

For example, from my pc (work) I only can connect to an instance specifying
an particular ip (the instance have 3) this ip is different from the one
that DNS resolves by default (if I use the name).

Even all instance are running on the default port, this we should always
concatenate ip (or name,  whatever people add) with port to connect.

Em 24/08/2016 13:32, "Rob Sewell" <notifications@github.com> escreveu:

**Comment by amtwo** [comment]
I don't think it's that unusual. I've worked in more than one place where there is at least one DB Server in a DMZ, or in another domain where explicitly including the port number was necessary to reliably connect. 

I've handled it in the past when doing similar data collection/monitoring by tacking the port onto the end of the FQDN (ex server.am2.co,54321), but I think it's more elegant to explicitly handle it by having the port number on InstanceList.  

Thinking thorough all the corner cases, I would go so far as to say InstanceList should include a column for the connection protocol. By default, we assume TCP (and check for port), but allow someone to specify that the connection happen over named pipes, too.
